### PR TITLE
feat: enable p2trMusig2 for testnet cold wallets

### PR DESCRIPTION
**Pull request by ericli-bitgo**
TICKET: BTC-1229

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts [hunk 1]
```diff
@@ -34,5 +34,9 @@ export function getUtxoCoinScriptTypesForWalletType(
   const scriptTypes = getUtxoCoinScriptTypes2Of3(coinName);
 
   // Only return true for p2trMusig2 if the wallet type is hot
```
Comment (rushilbg) [inline modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:36]: This comment would need to be updated as well.

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
this is getting hard to follow

can you add a function

```
function isEnabledAddressType(network: utxolib.Network, walletType: string, scriptType: ScriptType2Of3): boolean
```

File: modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts [hunk 2]
```diff
@@ -22,6 +22,19 @@ export function getUtxoCoinScriptTypes2Of3(coinName: string): utxolib.bitgo.outp
...
 
+/**
+ * Check if script type is enabled for a given walletType and network
+ * @param network
+ * @param walletType
+ * @param scriptType
+ */
+function isEnabledAddressType(network: utxolib.Network, walletType: WalletType, scriptType: ScriptType2Of3): boolean {
+  if (scriptType !== 'p2trMusig2') {
+    return true;
```
Comment (OttoAllmendinger) [inline modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:33]:
```suggestion
    return utxolib.isSupportedScriptType(network, scriptType);
```

File: modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts [hunk 3]
```diff
@@ -22,6 +22,19 @@ export function getUtxoCoinScriptTypes2Of3(coinName: string): utxolib.bitgo.outp
...
+  }
+  return walletType === 'hot' || (walletType === 'cold' && utxolib.isTestnet(network));
+}
```
Comment (OttoAllmendinger) [inline modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:36]:
```suggestion
function isEnabledAddressType(network: utxolib.Network, walletType: WalletType, scriptType: ScriptType2Of3): boolean {
  if (!utxolib.isSupportedScriptType(network, scriptType)) {
    return false;
  }
  if (scriptType === 'p2trMusig2') {
    return walletType === 'hot' || (walletType === 'cold' && utxolib.isTestnet(network));
  }
  return true;
}
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

**Review by OttoAllmendinger — DISMISSED** [review]

File: modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts [hunk 4]
```diff
@@ -33,6 +49,10 @@ export function getUtxoCoinScriptTypesForWalletType(
 ): utxolib.bitgo.outputScripts.ScriptType2Of3[] {
   const scriptTypes = getUtxoCoinScriptTypes2Of3(coinName);
 
-  // Only return true for p2trMusig2 if the wallet type is hot
-  return scriptTypes.filter((scriptType) => (scriptType === 'p2trMusig2' ? walletType === 'hot' : true));
+  const coin = coins.get(coinName);
+  assert(coin instanceof UtxoCoin, `coin ${coinName} is not a utxo coin`);
+  const network = utxolib.networks[coin.network.utxolibName as utxolib.NetworkName];
+
+  // Only return true for p2trMusig2 if the wallet type is hot or cold on testnet
```
Comment (OttoAllmendinger) [inline modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:56]:
```suggestion
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

**Review by OttoAllmendinger — APPROVED** [review]
